### ISSUE TITLE
Encoding rules

### ISSUE THREAD

**Issue by ghost** [nieltg/munemo-rs#1]
Is there any description on the encoding rules? To put more clearly, why do -1 corresponds to xabi, hayu to 2592, and so on?

Also, I see that you made this library to study Rust. Good luck on your journey learning Rust!

**Comment by nieltg** [comment]
Oh, I missed this one. The encoding rule is simple. As you can see, we have constants [here](https://github.com/nieltg/munemo-rs/blob/ea1fc5347cb4324fda5bbecae269bc384291a5da/src/constants.rs):
- `NEGATIVE_SYLLABLE`: syllable to represent a negative number.
- `SYLLABLES`: an array of syllables, corresponding to each number.

Every negative number will be encoded into a word that starts with `xa`, then the rest will be treated as a positive number.
The positive number itself will be encoded using division and modulo operation. I'll use 2592 as an example:
- Modulo of 2592 by 100 (length of `SYLLABLES` array) is 92. If we use this as an index for looking up a value in `SYLLABLES` array, we'll found `yu` syllable.
- The number is divided by 100, resulting in 25, which will be processed again.
- Modulo of 25 by 100 is 25, which is `ha` on `SYLLABLES` array.
- Conclusion: the number is encoded as `hayu`.

If the number is negative, it'll be `xahayu`. Simple, huh?

Anyway, thanks for that! It's one in my learning-list that I hadn't ever learned until now, I forced myself by writing this crate. 😅

**Comment by ghost** [comment]
I see, interesting. Did you make the rule yourself, or are you inspired by something?

Cool, working on a project is definitely one of the better ways to learn a language!

**Comment by nieltg** [comment]
No, it's based on munemo Ruby gem which is stated [here](https://github.com/nieltg/munemo-rs/tree/7b011be4b61a83551dd937728b9d03190d6d61dc#references) and as you can see, munemo Ruby gem itself is also based on mnemo library, which is written on C.
